src/components/application_manager/include/application_manager/state_controller_impl.h

```diff
@@ -152,12 +152,17 @@ class StateControllerImpl : public event_engine::EventObserver,
 
   template <typename UnaryFunction>
   void ForEachApplication(UnaryFunction func) const {
-    DataAccessor<ApplicationSet> accessor = app_mngr_.applications();
-    ApplicationSet::iterator it = accessor.GetData().begin();
-    for (; it != accessor.GetData().end(); ++it) {
-      ApplicationConstSharedPtr const_app = *it;
-      if (const_app) {
-        func(app_mngr_.application(const_app->app_id()));
+    ApplicationSet applications;
+    {
+      DataAccessor<ApplicationSet> accessor = app_mngr_.applications();
+      applications = accessor.GetData();
+    }
+
+    ApplicationSet::iterator it = applications.begin();
+    for (; it != applications.end(); ++it) {
+      auto app = *it;
+      if (app) {
+        func(app);
```

Comment:
Can we use range-based like it:
for (auto& app : applications) {      
      if (app) {
        func(app);
      }
    }

Reply:
@SNiukalov Changed in c874387b10610d059a863f43497779f73aaa26e7